File: includes/c/syscalls/syscalls0.c
#include <unistd.h>
#include <stdio.h>
#include <stdlib.h>
#include <limits.h>

int main(void) {

    char hostname[HOST_NAME_MAX + 1];

    if (gethostname(hostname, sizeof(hostname)) == 0 ) {
        printf("Hostname: %s\n", hostname);
    } else {
        perror ("gethostname");
        exit(1);
    }
    printf("Effective user: %d\n", geteuid());
    printf("Process ID: %d\n", getpid());
    printf("Parent process ID: %d\n", getppid());
    exit(0);
}

File: includes/rs/syscalls/syscalls0.rs
use nix::unistd::{geteuid,getpid,getppid,gethostname};

fn main()  {

    let hostname = gethostname().unwrap();
    let pid = getpid();
    let ppid = getppid();
    let euid = geteuid();

    println!("Hostname: {:?}", hostname);
    println!("Effective user: {}", euid);
    println!("Process ID: {}" , pid);
    println!("Parent process ID: {}", ppid);
}
